supernode/server/0.3_bridge.go

```diff
@@ -88,6 +88,7 @@ func (s *Server) registry(ctx context.Context, rw http.ResponseWriter, req *http
 		TaskURL:     request.TaskURL,
 		SupernodeIP: request.SuperNodeIP,
 	}
+	cutil.DefaultHTTPSClient.RegisterTLSConfig(taskCreateRequest.RawURL, request.Insecure, request.RootCAs)
```

Comment:
I think we should create a http client to interact with the source for supernode instead of use the `cutil.DefaultHTTPSClient` which will bring some potential risks for other callers. WDYT?